src/index.js

```diff
@@ -38,6 +47,10 @@ class PromisePool extends Emitter {
     }
 
     if (this._numActive >= this._numConcurrent) {
+      if (this._pending.length >= this._maxPending) {
+        // the pool is likely being used in a manner that does not propagate backpressure
+        throw new Error('cannot queue function in pool: please read the documentation');
```

Comment:
Similar here - I think calling out the documentation is nice but it would be nice to say that the reason you can't queue the function is because only `n` pending requests are allowed.

Reply:
does f66f280 address your concern, or do you think the specific number is important?